File: kobciye/lib/core/constants/role_redirect.dart
import 'app_roles.dart';
import 'app_routes.dart';

/// Maps a role to its dashboard home route.
String roleHomeRoute(String role) {
  switch (role) {
    case AppRole.superAdmin:
      return AppRoutes.superAdmin;
    case AppRole.schoolAdmin:
      return AppRoutes.schoolAdmin;
    case AppRole.teacher:
      return AppRoutes.teacher;
    case AppRole.accountant:
      return AppRoutes.accountant;
    case AppRole.parent:
      return AppRoutes.parent;
    case AppRole.student:
      return AppRoutes.student;
    default:
      return AppRoutes.login;
  }
}

File: kobciye-rn/src/navigation/roleRedirect.js
import { AppRole } from '../constants/roles';

// Maps a role to its dashboard screen name (registered in RootNavigator).
export function roleHomeScreen(role) {
  switch (role) {
    case AppRole.superAdmin: return 'SuperAdminDashboard';
    case AppRole.schoolAdmin: return 'SchoolAdminDashboard';
    case AppRole.teacher: return 'TeacherDashboard';
    case AppRole.accountant: return 'AccountantDashboard';
    case AppRole.parent: return 'ParentDashboard';
    case AppRole.student: return 'StudentDashboard';
    default: return 'Login';
  }
}
